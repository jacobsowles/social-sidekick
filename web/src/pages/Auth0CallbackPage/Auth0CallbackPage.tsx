import React, { ComponentState, PureComponent } from 'react';

interface Auth0CallbackPageProps {
  handleAuthentication: (nextState: ComponentState) => void;
}

class Auth0CallbackPage extends PureComponent<Auth0CallbackPageProps> {
  public componentDidMount() {
    this.props.handleAuthentication(this.props);
  }

  public render() {
    return <div className="auth0-callback-page" />;
  }
}

export default Auth0CallbackPage;
