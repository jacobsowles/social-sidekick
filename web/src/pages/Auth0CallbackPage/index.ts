import Auth0CallbackPage from './Auth0CallbackPage';
export default Auth0CallbackPage;
