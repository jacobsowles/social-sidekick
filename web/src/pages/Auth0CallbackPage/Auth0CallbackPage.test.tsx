import React from 'react';

import { init, shallow, shallowTopLevelElement, toJson } from '@tests/component-test-base';
import Auth0CallbackPage from '@pages/Auth0CallbackPage';

describe('Auth0CallbackPage', () => {
  init(() => <Auth0CallbackPage handleAuthentication={jest.fn} />, 'div');

  it('should render the component without crashing', () => {
    toJson(shallow()).should.matchSnapshot();
  });

  it('should only include the default class names', () => {
    shallowTopLevelElement()
      .prop('className')
      .should.equal('auth0-callback-page');
  });
});
