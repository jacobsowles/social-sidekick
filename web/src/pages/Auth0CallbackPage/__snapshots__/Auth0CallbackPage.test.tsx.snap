// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth0CallbackPage should render the component without crashing 1`] = `
<div
  className="auth0-callback-page"
/>
`;
